osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -94,6 +94,12 @@ public HitObject()
 
                 foreach (var nested in NestedHitObjects)
                     nested.StartTime += offset;
+
+                if (DifficultyControlPoint != DifficultyControlPoint.DEFAULT)
+                    DifficultyControlPoint.Time = time.NewValue;
+                // traditionally this used EndTime, but at the point changes are being made in the editor this should no longer be considered relevant.
+                if (SampleControlPoint != SampleControlPoint.DEFAULT)
+                    SampleControlPoint.Time = time.NewValue;
```

Comment:
As a side note we may eventually have to have `EndTimeBindable` to fix issues like https://github.com/ppy/osu/issues/13473. Out of scope here, just mentioning in passing.

Also, on comparing with `ApplyDefaults()` lower down, is there a reason why `control_point_leniency` is dropped here? I guess the assumption is that start time should be snapped to full milliseconds or something?

Reply:
i'm not sure which is correct at this point. hard to say without some intensive testing, which might be out of scope for now..

Reply:
This may be the reason for exports to stable not getting correct velocities applied, and needs further investigation before merge.

Reply:
As I mentioned before, end time is likely more correct here as hit samples are played when a hitobject ends and not when it starts. That being said I forgot how slider slide/whistle/spinner samples work.

Is there a reason for changing this? Isn't it easy to `GetEndTime() + control_point_leniency` as before?

I would rather not change things because "i'm not sure which is correct at this point" when we've resolved many issues over the years to get hitsounds on-par with stable.

Reply:
Yep, I'll definitely try changing it back. The issue is that the only part which this affects is playback on stable, mostly in cases which we haven't encountered and don't have tests for yet.

Picture the scenario where in lazer, a mapper assigns a sample to a tick, which is encoded out to be `tick.EndTime+allowance` as a global `SampleControlPoint`. It's feasible that this could be read incorrectly by stable, potentially applying to the hitobject itself (if it occurs towards to end time of the parent object) I think? May also be why stable uses different values for the lenience for different lookups.

Either way, this is part of the remaining tests that are required to get this PR merged. A similar issue is confirmed to already be occurring for SV changes, as I've mentioned below.